src/scaffoldmaker/meshtypes/meshtype_3d_lung2.py

```diff
@@ -0,0 +1,2139 @@
+            'Accessory lobe ventral width',
+            'Accessory lobe medial curvature',
+            'Diaphragm centre x',
+            'Diaphragm centre y',
+            'Diaphragm curvature x',
+            'Diaphragm curvature y',
+            'Rotation in degree about [z, y, x]: Left lung',
+            'Rotation in degree about [z, y, x]: Right lung',
+            'Rotation in degree about [z, y, x]: Accessory lobe',
+            'Refine',
+            'Refine number of elements'
+        ]
+        return optionNames
+
+    @classmethod
+    def checkOptions(cls, options):
+        '''
+        :return:  True if dependent options changed, otherwise False.
+        '''
+        dependentChanges = False
+        if options['Refine number of elements'] < 1:
+            options['Refine number of elements'] = 1
+        if options['Number of left lung lobes'] > 2:
+            options['Number of left lung lobes'] = 2
+        if len(options['Rotation in degree about [z, y, x]: Left lung']) != 3:
+            options['Rotation in degree about [z, y, x]: Left lung'] = [0.0, 0.0, 0.0]
+        if len(options['Rotation in degree about [z, y, x]: Right lung']) != 3:
+            options['Rotation in degree about [z, y, x]: Right lung'] = [0.0, 0.0, 0.0]
+        if len(options['Rotation in degree about [z, y, x]: Accessory lobe']) != 3:
+            options['Rotation in degree about [z, y, x]: Accessory lobe'] = [0.0, 0.0, 0.0]
+        return dependentChanges
+
+    @classmethod
+    def generateBaseMesh(cls, region, options):
+        '''
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: annotationGroups
+        '''
+        # Generate two meshes: geometric[0] and lung[1] coordinates
+        for coordinate in range(2):
+            if coordinate == 0:
+                fm_n0 = region.getFieldmodule()
+                coordinates_n0 = findOrCreateFieldCoordinates(fm_n0)
+                # point reference
+                fm = fm_n0
+                coordinates = coordinates_n0
+            else:
+                region_n1 = region.createRegion()
+                fm_n1 = region_n1.getFieldmodule()
+                coordinates_n1 = findOrCreateFieldCoordinates(fm_n1, name="lung coordinates")
+                # point reference
+                fm = fm_n1
+                coordinates = coordinates_n1
+                # update material coordinates according to geometric parameters
+                options['Material Parameters']['Number of left lung lobes'] = options['Number of left lung lobes']
+                options['Material Parameters']['Open fissures'] = options['Open fissures']
+                options['Material Parameters']['Accessory lobe'] = options['Accessory lobe']
+                options = options['Material Parameters']
```

Comment:
A simpler and more efficient way to do this is to make the "coordinates" in the shape of the material coordinates first (since your transformations start from that shape), then do the following:

1. temporarily rename "coordinates" to "lung coordinates" (requires lung coordinates to not be defined yet).
   `coordinates.setName("lung coordinates")`
2. write the model to memory
   `# set up sir with a memory resource then:`
   `region.write(sir)`
3. Rename coordinates back:
   `coordinates.setName("coordinates")`
4. read the memory buffer
   `# set up sir2 with a memory buffer resource as written earlier:`
   `region.read(sir2)`
5. Find material coordinate field if needed later, e.g. as part of your functions:
   `material_coordinates = fm.findFieldByName("lung coordinates").castFiniteElement()`
6.  Now apply your transformation functions to the coordinates field only.

Reply:
I ended up having parameterised scaffold in the 'material coordinates'. I am not sure if this is what you are looking for